src/main/java/io/hops/hopsworks/expat/migrations/featurestore/statistics/StatisticsMigration.java

```diff
@@ -253,15 +266,34 @@ private void migrateFeatureDescriptiveStatistics(ResultSet fdsResultSet)
             windowEndCommitTime = commitTime; // for non-time-travel-enabled fgs, set end window as committime
           }
           // migrate fg stats
-          migrateFeatureGroupStatistics(statisticsId, filePath, windowStartCommitTime, windowEndCommitTime,
-            insertFdsStmt, insertFgFdsStmt);
-          // set window start commit time if time travel-enabled fg
-          if (updateFeatureGroupStatisticsCommitWindow(updateFgsStmt, statisticsId, windowStartCommitTime)) {
-            updateFgStatistics = true;
+          boolean fdsInserted = migrateFeatureGroupStatistics(statisticsId, filePath, windowStartCommitTime,
+            windowEndCommitTime, insertFdsStmt, insertFgFdsStmt);
+          
+          if (fdsInserted) {
+            // set window start commit time if time travel-enabled fg
+            if (updateFeatureGroupStatisticsCommitWindow(updateFgsStmt, statisticsId, windowStartCommitTime)) {
+              updateFgStatistics = true;
+            }
+          } else {  // if fds not inserted
+            // this fg statistics could not be migrated to DB, so we delete the fg stats row.
+            LOGGER.info(String.format(
+              "[migrateFeatureDescriptiveStatistics] -- marking fg statistics for deletion, with id '%s' and " +
+                "feature group id '%s'", statisticsId, entityId));
+            deleteFeatureGroupStatistics(deleteFgsStmt, statisticsId);
```

Comment:
If we're here it's because `fdsInserted` is false, which means we failed to insert. In this case what do we delete here?